src/thorin/pass/fp/eta_red.h

```diff
@@ -9,8 +9,8 @@ namespace thorin {
 /// Rewrites `λx.e x` to `e`, whenever `x` does (optimistically) not appear free in `e`.
 class EtaRed : public FPPass<EtaRed> {
 public:
-    EtaRed(PassMan& man)
-        : FPPass(man, "eta_red") {}
+    EtaRed(PassMan& man, bool callee_only = false)
+        : FPPass(man, "eta_red"), callee_only_(callee_only) {}
 
```

Comment:
Essentially the closure-destruction phase assumes that code-part of a closure is always a Lambda. This flag is used to prevent EtaRed from reducing inside closures.